.github/wiki/Project-Configurations.md

```diff
@@ -0,0 +1,136 @@
+
+- Debug
+- Release
+
+However, these basic configurations are not enough for our development process. Since our team always wants to ensure the application is working in a designated manner, additional levels of testing are required before the application is ready to be used.
+
+The recommended set of configurations to have is:
+
+- Debug Staging
+- Release Staging
+- Debug Production
+- Release Production
+
+### Active Compilation Conditions
+
+Having distinct flags for each configuration is the preferred solution.
+
+Xcode provides a build setting named `Active Compilation Conditions`, which supports passing conditional compilation flags to the Swift compiler.
+
+In our iOS template project, there are three custom conditional compilation flags: `DEBUG`, `STAGING`, `PRODUCTION`. The following table will describe how the developers differentiate a configuration from the others.
+
+| Build Configurations | DEBUG | STAGING | PRODUCTION |
+|---|---|---|---|
+| Debug Staging | ✔︎ | ✔︎ |  |
+| Release Staging |  | ✔︎ |  |
+| Debug Production | ✔︎ |  | ✔︎ |
+| Release Production |  |  | ✔︎ |
+
+The huge advantage when using custom flags is customizable specific features based on a particular environment.
+
+Example: 
+
+Specify a value based on the environment:
+
+```swift
+enum Environment {
+
+    static func based<T>(staging: T, production: T) -> T {
+        #if PRODUCTION
+        return production
+        #else
+        return staging
+        #endif
+	}
+}
+```
+
+During the development process, all functionalities and frameworks are available in the application. However, there can be situations when the developer turns on some features for a particular environment without creating separate modules for the application. In these cases, taking advantage of `Active Compilation Conditions` by declaring a condition to include or exclude code is the recommended solution.
+
+```swift
+#if DEBUG
+    // Code the app includes in DEBUG environments
+#else
+    // Code the app includes when it is not build with DEBUG environments
+#endif
+```
+
+### Xcode Configuration **File**
+
+A Configuration Settings File (`*.xcconfig`), also known as a build configuration file, is a plain text file that defines and overrides the build settings for a particular build configuration of a project or target. This type of file can be edited outside of Xcode and integrates well with source control systems.
+
+Because of different settings for each environment (such as bundle identifiers, endpoints, etc.), the developers also define them in different *.xcconfig files.
+
+### **Settings Bundle**
+
+To support multiple levels of testing, the `Settings Bundle` is usually integrated along with the dev configurations. In addition, the Settings Bundle facilitates the testing process. It creates a shortcut to change environment values without using any third-party API or server.
```

Comment:
Please specify what type of files is this ie: .plist, etc.

Reply:
Fixed in https://github.com/nimblehq/ios-templates/pull/268/commits/4ac43d42bd69de88234f85aee75cd8736fdfc56f